src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx

```diff
@@ -222,6 +222,31 @@ function BarplotViz(props: VisualizationProps) {
       ? data.value?.completeCasesAllVars
       : data.value?.completeCasesAxesVars;
 
+  // find dependent axis max value
+  const defaultDependentMaxValue = useMemo(() => {
+    return data?.value?.series != null
+      ? Math.max.apply(
+          Math,
+          data?.value?.series.map((o) => {
+            let localMax = Math.max(...o.value);
+            console.log('local max =', localMax);
+            return localMax;
+          })
+        )
+      : undefined;
+  }, [data, data.value, variable, overlayVariable]);
```

Comment:
Don't need both `data` and `data.value` ?

Reply:
Elsewhere we've used flatMap and lodash Max, I think?

Reply:
@bobular thank you for your review! As for the scatterplot fix, honestly I don't know if it is necessary so your input would be helpful. Here is the situation I have faced for that. When I checked one of my favorite test variables (haha) for scatterplot viz, Sleeping rooms in dwelling (X) & Persons sleeping in dwelling count, I realized that the type of Persons sleeping in dwelling count is now classified as integer (previously number). default-dependent-axis-range.ts only checks number or date so if integer type comes in, no (dependent axis) range is set. I thought that such an integer type should also have the (fixed) range like number. The change is related to this. Do you think this is reasonable to change ?

As for data and data.value, yeah actually one of them works. Perhaps data will be better: will change it